server/code/setup/cpsetup

```diff
@@ -118,6 +131,21 @@ class TomcatSetup(object):
         config.write(updated)
         config.close()
 
+        # Edit tomcat.conf
+        original_main_file = open(self.main_conf, 'r')
+        original_main = original_main_file.read()
+        original_main_file.close()
+
+        if re.search(self.main_existing_pattern, original_main, re.MULTILINE):
+            updated_main = self._replace_current_main(original_main)
+        elif re.search(self.main_comment_pattern, original_main, re.MULTILINE):
+            updated_main = self._replace_commented_main(original_main)
+        else:
+            updated_main = original_main + os.linesep + self.new_java_home
+        main_config = open(self.main_conf, 'w')
```

Comment:
You could use Python's with keyword to close the file automatically.
```
with open(self.main_conf, 'w') as main_config:
    main_config.write(updated_main)
```

Reply:
updated